game/script-greetings.rpy

```diff
@@ -10,14 +10,28 @@ init python in greetings:
         """
         Picks a random greeting, accounting for affinity
         """
-        # Get the farewells the current affinity allows for us
-        #TODO: Generalized filter function
-        return random.choice(
-            store.Topic.filter_topics(
-                GREETING_MAP.values(),
-                affinity=store.jn_globals.current_affinity_state
-            )
-        ).label
+        # If the player didn't make an admission that had Natsuki make them leave, pick a random greeting
+        # based on the player's affinity
+        if not store.persistent.jn_player_admission_type_on_quit:
+            return random.choice(
+                store.Topic.filter_topics(
+                    GREETING_MAP.values(),
+                    affinity=store.jn_globals.current_affinity_state,
+                    excludes_categories=["Admission"]
+                )
+            ).label
+
+        # Otherwise, pick a greeting that matches the admission made
+        else:
+            return random.choice(
+                store.Topic.filter_topics(
+                    GREETING_MAP.values(),
+                    affinity=store.jn_globals.current_affinity_state,
+                    additional_properties=[
+                        ("admission_type", store.persistent.jn_player_admission_type_on_quit),
+                    ]
+                )
+            ).label
```

Comment:
(This can be cleaned up a little further) but I think it'll be best to use kwargs for this so we can avoid restating the code more or less twice here.

```suggestion
        _kwargs = dict()
        
        _kwargs.update(
            {"excludes_categories": ["Admission"]} if not store.persistent.jn_player_admission_type_on_quit else {"additional_properties": [("admission_type", store.persistent.jn_player_admission_type_on_quit)]}
        )

        return random.choice(
            store.Topic.filter_topics(
                GREETING_MAP.values(),
                affinity=store.jn_globals.current_affinity_state,
                **kwargs
            )
        ).label
```

Additionally, we should probably reset the variable for the last admission on quit here as it seems like this will begin to span over multiple sessions of the player doesn't admit to anything else.